ntbs-service/Services/AlertService.cs

```diff
@@ -118,22 +118,39 @@ public async Task DismissAllOpenAlertsForNotification(int notificationId)
 
             foreach (var alert in alerts)
             {
+                // notification may have a duplicate alert to be dismissed on multiple other records
                 if (alert is DataQualityPotentialDuplicateAlert duplicateAlert)
                 {
-                    var duplicate =
-                        await _notificationRepository.GetNotificationForAlertCreationAsync(duplicateAlert.DuplicateId);
+                    await DismissDuplicationAlertsOnDuplicateRecord(notificationId, duplicateAlert.DuplicateId);
+                }
+                
+                CloseAlert(alert);
+            }
 
-                    if (duplicate.NotificationStatus.IsOpen())
+            await _alertRepository.SaveAlertChangesAsync();
+        }
+
+        private async Task DismissDuplicationAlertsOnDuplicateRecord(int notificationId, int duplicateAlertId)
+        {
+            var duplicateRecordAlerts = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
+            if (duplicateRecordAlerts != null)
+            {
+                foreach (var alert in duplicateRecordAlerts)
+                {
+                    if (alert is DataQualityPotentialDuplicateAlert duplicateAlert && 
+                        duplicateAlert.DuplicateId == notificationId)
```

Comment:
I would move this condition into a where clause when we fetch the alerts to avoid nesting. Something like: 
`.where(al => al is DataQualityPotentialDuplicateAlert && al.DuplicateId == notificationId)`